cfe_ts_crc.c

```diff
@@ -125,30 +127,43 @@ int main(int argc, char **argv)
     if (fd < 0)
     {
         printf("\ncfe_ts_crc error: can't open input file!\n");
-        exit(0);
+        printf("%s\n", strerror(errno));
+        exit(1);
     }
     /* seek past the number of bytes requested */
-    lseek(fd, skipSize, SEEK_SET);
+    offsetReturn = lseek(fd, skipSize, SEEK_SET);
+    if (offsetReturn != skipSize)
+    {
+        printf("\ncfe_ts_crc error: lseek failed!\n");
+        printf("%s\n", strerror(errno));
+        exit(1);
+    }
 
     /* read the input file 100 bytes at a time */
-    while (done == 0)
+    do
     {
         readSize = read(fd, buffer, 100);
-        fileCRC  = CalculateCRC(buffer, readSize, fileCRC);
+        if (readSize < 0)
+        {
+            printf("\ncfe_ts_crc error: file read failed!\n");
+            printf("%s\n", strerror(errno));
+            exit(1);
+        }
+        fileCRC = CalculateCRC(buffer, readSize, fileCRC);
         fileSize += readSize;
-        if (readSize != 100)
-            done = 1;
-    }
+    } while (readSize != 100);
```

Comment:
Problem here is this isn't fixing the EOF check in #41.  It is the same test just done in a do/while construct instead of a while.  But it's the same thing, still has the same issue.

I would typically expect something more like:

```
while (true)
{
    readSize = read(fd, buffer, sizeof(buffer));
    if (readSize == 0)
    {
        break;
    }
    if (readSize < 0)
    {
        perror("read");
        exit(1);
    }
    fileCRC = CalculateCRC(...);
    fileSize += readSize;
}

Reply:
yeah leaving that as 100 just slipped through.